cachi2/core/package_managers/gomod.py

```diff
@@ -448,9 +502,16 @@ def _find_missing_gomod_files(source_path: RootedPath, subpaths: list[str]) -> l
     return invalid_gomod_files
 
 
-def _resolve_gomod(
-    app_dir: RootedPath, request: Request, tmp_dir: Path
-) -> tuple[ParsedModule, Iterable[ParsedModule], Iterable[ParsedPackage]]:
+class ResolvedGoModule(NamedTuple):
```

Comment:
Nitpick: I'd still move this class to the top of the file alongside the other class definitions (although there is still one class we carried over from Cachito that is defined in the middle of functions :disappointed: ).

And the `Go` part of the name is probably redundant.